core/src/main/java/hudson/model/AsyncAperiodicWork.java

```diff
@@ -40,8 +40,48 @@
  * @since 1.410
  */
 public abstract class AsyncAperiodicWork extends AperiodicWork {
-	
-	/**
+    /**
+     * The default number of minutes after which to try and rotate the log file used by {@link #createListener()}.
+     * This value is controlled by the system property {@code hudson.model.AsyncAperiodicWork.logRotateMinutes}.
+     * Each individual AsyncAperiodicWork can also have a per-extension override using the system property
+     * based on their fully qualified class name with {@code .logRotateMinutes} appended.
+     *
+     * @since 1.650
+     */
+    private static final long LOG_ROTATE_MINUTES = Long.getLong(AsyncAperiodicWork.class.getName() +
+            ".logRotateMinutes", TimeUnit.DAYS.toMinutes(1));
+    /**
+     * The default file size after which to try and rotate the log file used by {@link #createListener()}.
+     * A value of {@code -1L} disables rotation based on file size.
+     * This value is controlled by the system property {@code hudson.model.AsyncAperiodicWork.logRotateSize}.
+     * Each individual AsyncAperiodicWork can also have a per-extension override using the system property
+     * based on their fully qualified class name with {@code .logRotateSize} appended.
+     *
+     * @since 1.650
```

Comment:
@KostyaSha the point is that `master` is supposed to be the next version. If @daniel-beck as security lead has decreed that 1.650 shall be just 1.649 + security fixes, then the version number in `master` should have been adjusted accordingly.

The habit of using `@since TODO` or `@since FIXME` is a bad pattern as there are countless examples of them not actually being corrected for the merge... I now think it is much better to be explicit in the PR and complain if the PR is saying the wrong version against the branch that it will be merged into... which says that either @daniel-beck will need to fix up any `@since 1.650` versions when the 1.650 is being rolled... or @daniel-beck needs to bump the version on `master` to `1.651-SNAPSHOT` and then I need to fix my `@since` tags before merge